docs/management_api.yml

```diff
@@ -28,83 +28,80 @@ tags:
           description: OK
           content:
             application/json:
               schema:
-                type: array
-                items:
-                  $ref: '#/components/schemas/Parameter'
+                type: object
+                properties:
+                  id:
+                    type: string
+                  expected:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Parameter'
+                  actual:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Parameter'
+                  updated_ts:
+                    type: string
+                  report_ts:
+                    type: string
         400:
-          description: Invalid Request.
+          description: Bad Request.
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
-        500:
-          description: Internal Server Error.
+        404:
+          description: Not Found.
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
-
     put:
-      operationId: Set Configuration
+      operationId: Store a set of key-value pairs
       tags:
         - ManagementAPI
-      summary: Assign configuration parameters
+      summary: Set a key-value pair store, updating if existing, removing if empty
       requestBody:
         content:
           application/json:
             schema:
-              oneOf:
-                - $ref: '#/components/schemas/Parameter'
-                - type: array
+              type: object
+              properties:
+                expected:
+                  type: array
                   items:
                     $ref: '#/components/schemas/Parameter'
-      responses:
-        200:
-          description: Parameters set and updated successfully.
-        400:
-          description: Invalid Request.
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
-        500:
-          description: Internal Server Error.
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
-
-    delete:
-      operationId: Unset Configuration
-      tags:
-        - ManagementAPI
-      summary: Assign configuration parameters
       parameters:
-        - name: 'key'
-          in: 'query'
+        - in: path
+          name: deviceId
           schema:
-            description: Name of the parameter(s) to delete.
-            type: "array"
-            items:
-              type: string
+            type: string
+          required: true
+          description: ID of the device to query.
       responses:
-        200:
-          description: Parameters set and updated successfully.
+        201:
+          description: Created
```

Comment:
201 is not correct, because this method is an upsert. To make it easier to use, though, the PUT should return 200 and the updated device configuration (full object). This simplifies the consumption of the end-point by the UI, for example.

Reply:
204 it is.

Reply:
204 is way better, thanks. I still believe it would be useful to return the updated object, though, instead of an empty response. I won't fight for this, take it as a suggestion.